projects/observability/src/shared/components/explore-query-editor/explore-visualization-builder.ts

```diff
@@ -117,18 +127,44 @@ export class ExploreVisualizationBuilder implements OnDestroy {
   }
 
   private buildRequest(state: ExploreRequestState): ExploreVisualizationRequest {
+    const orderBy = this.getOrderBy(state.series[0], state.interval, state.orderBy);
```

Comment:
@aaron-steinfeld I do not totally agree with this approach but I should apply the new orderBy to the attribute and also to the exploreQuery$ to remove the param or add it when it's necessary. It's working fine but any suggestion here?

Reply:
Not sure I'm following why this is separate. The order by should already be part of the state. Let me read on.

Reply:
My understanding after reading it is we're using this function to set defaults, but we shouldn't be because this only translates the UI to a data request and we want the defaults to show in the actual UI - so that should have already happened at this point. It looks like there's two behaviors: clearing order by if interval is set and setting the default to the value of the current first series.

But we previously did the clearing immediately when interval was set, so why the change? This had been brought up in https://github.com/hypertrace/hypertrace-ui/pull/1747#discussion_r971068145 and the concern is that we want to make sure whenever interval is cleared, a default order is set, but when an interval is assigned, the order is cleared. That concern still holds. In this PR, we assign the default in two places - when decoding the URL (only if interval is set to none, so this only is relevant in restoring particular URL patterns), and when sending the query. So we've added that logic on the request building side, but not on the UI state I believe? How does the state (and thus the UI) get updated with the default?

For example - what's the result if I load explorer with the default state (calls time series, interval AUTO), then change the interval to None? The default order by was undefined, so my expectation from reading the code (and I certainly could have missed it), would that the query would be right, the UI order by selector would be wrong.

Reply:
Will try to pull this to test it out and see if I'm misunderstanding.

Reply:
It does appear to work, will try to understand the code better.

Reply:
There's some funkiness going on, but it may not be on the UI side. Easier to chat.

Reply:
fixed with workaround